test/env/getter.go

```diff
@@ -46,3 +51,84 @@ func GetIPsFromVMIStatus(vmi *kubevirtv1.VirtualMachineInstance, networkInterfac
 	}
 	return ifaceStatus.IPs
 }
+
+func virtualMachineInstancePod(vmi *kubevirtv1.VirtualMachineInstance) (*corev1.Pod, error) {
+	pod, err := lookupPodBySelector(vmi.Namespace, vmiLabelSelector(vmi), vmiFieldSelector(vmi))
+	if err != nil {
+		return nil, fmt.Errorf("failed to find pod for VMI %s (%s)", vmi.Name, string(vmi.GetUID()))
+	}
+	return pod, nil
+}
+
+func lookupPodBySelector(namespace string, labelSelector, fieldSelector map[string]string) (*corev1.Pod, error) {
+	pods := &corev1.PodList{}
+	err := Client.List(
+		context.Background(),
+		pods,
+		client.InNamespace(namespace),
+		client.MatchingLabels(labelSelector),
+		client.MatchingFields(fieldSelector))
+	if err != nil {
+		return nil, err
+	}
+
+	if len(pods.Items) == 0 {
+		return nil, fmt.Errorf("failed to lookup pod with labels %v, fields %v in namespace %s", labelSelector, fieldSelector, namespace)
+	}
+
+	return &pods.Items[0], nil
+}
+
+func vmiLabelSelector(vmi *kubevirtv1.VirtualMachineInstance) map[string]string {
+	return map[string]string{kubevirtv1.CreatedByLabel: string(vmi.GetUID())}
+}
+
+func vmiFieldSelector(vmi *kubevirtv1.VirtualMachineInstance) map[string]string {
+	fieldSelectors := map[string]string{}
+	if vmi.Status.Phase == kubevirtv1.Running {
+		const podPhase = "status.phase"
+		fieldSelectors[podPhase] = string(corev1.PodRunning)
+	}
+	return fieldSelectors
+}
+
+func parsePodNetworkStatusAnnotation(podNetStatus string) ([]nadv1.NetworkStatus, error) {
+	if len(podNetStatus) == 0 {
+		return nil, fmt.Errorf("network status annotation not found")
+	}
+
+	var netStatus []nadv1.NetworkStatus
+	if err := json.Unmarshal([]byte(podNetStatus), &netStatus); err != nil {
+		return nil, err
+	}
+
+	return netStatus, nil
+}
+
+func defaultNetworkStatus(vmi *kubevirtv1.VirtualMachineInstance) (*nadv1.NetworkStatus, error) {
+	virtLauncherPod, err := virtualMachineInstancePod(vmi)
+	if err != nil {
+		return nil, err
+	}
+
+	netStatuses, err := parsePodNetworkStatusAnnotation(virtLauncherPod.Annotations[nadv1.NetworkStatusAnnot])
+	if err != nil {
+		return nil, err
+	}
+
+	for _, netStatus := range netStatuses {
+		if netStatus.Default {
+			return &netStatus, nil
+		}
+	}
+	return nil, fmt.Errorf("primary IPs not found")
+}
+
+func IPsFromNetworkStatusAnnotation(vmi *kubevirtv1.VirtualMachineInstance) ([]string, error) {
+	defNetworkStatus, err := defaultNetworkStatus(vmi)
+	if err != nil {
+		return nil, err
+	}
+
+	return defNetworkStatus.IPs, nil
+}
```

Comment:
Move this function to the test, and call it `defaultNetworkStatusAnnotationIPs` so the test reads 
```golang 
ipsFrom: defaultNetworkStatusAnnotationIPs
```

Reply:
[DONE](https://github.com/kubevirt/ipam-extensions/compare/c2d54f708f9d71cc0e559422d635b7d03a4f9c0f..169064a4a7369d968bacf312b89914738c74c16d)